From 87818ef325c6103bdb5d1e84ff7073d5ea6d557b Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 27 Sep 2017 02:25:32 +0200
Subject: [PATCH] fw/base: Enable home button wake

Change-Id: I032843cf26d5e7847f90ae1f2fd06bb680f8870f
---

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 4477e5a..922d4ae 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -6042,6 +6042,12 @@
                 break;
             }
 
+            case KeyEvent.KEYCODE_HOME:
+                if (down && !interactive) {
+                    isWakeKey = true;
+                }
+                break;
+
             case KeyEvent.KEYCODE_ENDCALL: {
                 result &= ~ACTION_PASS_TO_USER;
                 if (down) {
